### PR TITLE
RELATED: BB-1108 Fix measure title & upgrade dependency

### PR DESCRIPTION
**Pull request by martinmosko**
Fixed algorithm for building measure title for derived measures from arithmetic measures

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2693/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2076/ : SUCCESS in 11m 46s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1106/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2696/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/653/ : UNSTABLE in 21m 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2697/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2079/ : SUCCESS in 8m 32s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1108/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2698/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/654/ : UNSTABLE in 19m 20s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2701/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2082/ : SUCCESS in 2m 36s

**Comment by martinmosko** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1109/artifact/test-storybook/html-report/index.html

File: src/helpers/measureTitleHelper.ts [hunk 1]
```diff
@@ -56,113 +43,179 @@ function findOverTimeComparisonType(measureDefinitionType: IMeasureDefinitionTyp
...
-) {
+): string {
     const masterMeasureIdentifier = getMasterMeasureIdentifier(measureDefinitionType);
     if (masterMeasureIdentifier === null) {
         return undefined;
     }
-    const masterMeasure = getMasterMeasure(measureBucketItems, masterMeasureIdentifier);
-    const derivedMeasureTitleBase = getDerivedMeasureTitleBase(masterMeasure);
+
+    const measureProps = findMeasureTitleItem(measureTitleProps, masterMeasureIdentifier);
```
Comment (xMort) [inline src/helpers/measureTitleHelper.ts]: This method can return null but the result is used without checking for the nullability.

File: src/helpers/tests/measureTitleHelper.spec.ts [hunk 2]
```diff
@@ -125,6 +134,17 @@ describe('measureTitleHelper', () => {
             );
         });
 
+        it('should ignore title attribute when its included in derived or arithmetic measure (computed title)', () => {
```
Comment (xMort) [inline src/helpers/tests/measureTitleHelper.spec.ts]: Typo: `its` => `it's` (or better `it is`)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2703/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2083/ : SUCCESS in 8m 00s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2705/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2085/ : SUCCESS in 7m 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2706/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2086/ : SUCCESS in 8m 06s

**Comment by xMort** [comment]
Lgtm

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/585/ : SUCCESS in 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2087/ : SUCCESS in 3m 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1110/ : SUCCESS in 28m 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/555/ : SUCCESS in 30m 42s

**Pull request merged**